tests/components/blue_current/__init__.py

```diff
@@ -19,35 +19,21 @@
 }
 
 
-async def connect(client, token: str):
-    """Set the connected event."""
-    if client.connect_exception:
-        # Raise exception once.
-        temp = client.connect_exception
-        client.connect_exception = None
-        raise temp
-
-    client.connected.set()
-    client.connected.clear()
-
-
-async def start_loop(client, receiver):
+async def start_loop(hass: HomeAssistant, client, receiver):
     """Set the receiver."""
     client.receiver = receiver
-    client.loop_start_task.set()
 
-    while True:
-        if client.loop_exception:
-            # Raise exception once.
-            temp = client.loop_exception
-            client.loop_exception = None
-            raise temp
-        await sleep(0)
+    client.started_loop.set()
+    client.started_loop.clear()
+
+    client.loop_future = hass.loop.create_future()
```

Comment:
I'd define a pytest fixture that creates the future and returns it. Use that fixture in the other fixtures or tests as needed.

We also then don't need to store the future on the client where it doesn't really belong.